python/cuda_cccl/cuda/cccl/_version_utils.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+CUDA version detection utilities shared across the cccl package.
+"""
+
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+
+def detect_cuda_version() -> Optional[int]:
```

Comment:
This function can just use `get_cuda_path` to get the path, and then call `get_cuda_version_from_path` to get the version.